templates.dist/fields.php

```diff
@@ -248,96 +248,55 @@ function renderRegionalAddition(string $inputName, array $selectedValue, bool $m
 	function renderParkingLot(array $parkingArray, string $language, string $locale, string $codeCurrency, string $currency): array
 	{
 		$messages = [];
-		foreach ($parkingArray as $key => $parking) {
-		$MarketingType = '';
-		$pluralPaking = '';
-		$preposition = "at";
-			if (!$parking['Count']) {
+		
+
+		foreach ( $parkingArray as $key => $parking ) {
+			$pluralPaking  = '';
+			$preposition   = "at";
+			$MarketingType = $parking['MarketingType'];
+			$parkingName='';
+			if ( ! $parking['Count'] ) {
 				continue;
 			}
 			if ( $parking['Count'] != 1 ) {
 				$pluralPaking = " each";
 			}
-			if ($codeCurrency == "EUR" && $locale == "de_DE" ) {
-				$preposition = "à";
-				$pluralPaking = "";
+
+			$parkingName = getParkingName( $key, $parking['Count'] );
+
+			if ( $codeCurrency == "EUR" && $locale == "de_DE" ) {
+				$MarketingType = ucwords( $parking['MarketingType'] );
+				$parkingName   = ucwords( $parkingName );
+				$preposition   = "à";
+				$pluralPaking  = "";
 			}
-			if (!empty($parking['MarketingType'])) {
-				$MarketingType = (' (' . $parking['MarketingType'] . ') ');
+			if ( ! empty( $parking['MarketingType'] ) ) {
+					$MarketingType = ( ' (' . $MarketingType . ')' );
 			}
-			/* translators: 1: Name of parking lot, 2: Price , 3: Each , 4: MarketingType */
-			$element = sprintf( __( '%1$s ' . $preposition . ' %2$s' . '%3$s' . '%4$s', 'onoffice' ), getParkingName($key, $parking['Count']), formatPriceParking($parking['Price'], $language, $locale, $codeCurrency, $currency), $pluralPaking, $MarketingType );
-			
-			array_push($messages, $element);
+			/* translators: 1: Name of parking lot, 2: Preposition, 3: Price, 4: Each, 5: MarketingType */
+			$element = sprintf( __( '%1$s ' . '%2$s' . ' %3$s' . '%4$s' . '%5$s', 'onoffice' ), $parkingName,
```

Comment:
This will not be translatable (see #317). String concatenations are not recognized and you are using variables where you could just place it into the translation string.

There is a complication because we need to translate the parking name differently, depending on whether it is singular or plural.

One possibility:

```php
if ($key == "garage") {
  $countAndName = _n("%s garage", "%s garages", $count);
} else if ($key == "multiStoryGarage") {
  $countAndName = _n("%s multi-story garage", "%s multi-story garages", $count);
} else if ...

$marketingTypeOutput = "";

$marketingType = $parking['MarketingType'];
if ( ! empty($marketingType) ) {
  if ($marketingType == "rent") {
    $marketingTypeOutput = __(" (rent)", 'onoffice-for-wp-websites');
  } else if ($marketingType == "buy") {
    $marketingTypeOutput = __(" (purchase)", 'onoffice-for-wp-websites');
  } else {
    // Throw an error about unrecognized marketing type.
  }
}

$price = formatPriceParking( $parking['Price'], $language, $locale, $codeCurrency, $currency );

/* translators: 1: Count and name of parking lot, 2: Price, 3: Marketing type */
$element = sprintf( _n( '%1$s at %2$s%3$s', '%1$s at %2$s each%3$s', $count, 'onoffice-for-wp-websites' ), $countAndName, $price, $marketingTypeOutput);